### DISCUSSION TITLE
Snapshot on a Ding Event?

### DISCUSSION THREAD

**Discussion by Pabsilon** [tsightler/ring-mqtt#691] [Support Q&A]
Hi! I've been using ring-mqtt for I'd say more than 4 years now, and I'm very happy with it.
I recently moved to a house instead of a flat, and I was able to install a doorbell. It's a battery powered doorbell, and I'm in the EU, so interval snapshots or motion snapshots are a big no-no. In my country, at least, there is an exception for recording on video doorbells after someone rings, just to 'identify' who called. Basically the way the ring app works.

I've been experimenting with ding notifications on my phone, but I was wondering if there's any way I could grab a snapshot, or the same snapshot that the ring app shows on my phone to have it in HA? That would make for more interesting phone and tv notifications, and only show a single notification.

I noticed that snapshots could be taken on motion and on interval (or both), but I always wondered why we couldn't have 'on ding' snapshots.

Thanks!

**Comment by tsightler** [comment]
So I'm curious, are you saying that in your country you don't get alerts from the Ring app for motion in front of your doorbell, only if someone actually rings the doorbell?  So if you open the Ring app you don't see previews of the current image on the camera?  Also, if somebody comes up and rings the doorbell you don't get a recording of them walking up?

I guess I'd say that the reason it's never come up is because Ring devices are marketed in the US primarily as security cameras and as a defense against package theft, and porch pirates don't ring your doorbell.  Heck, many delivery drivers don't ring the doorbell, but one of the key features of Ring is to detect package deliveries.

Not only that, but battery Ring cameras (the most common type by far because they are cheap) can't take snapshots while recording, but in pretty much every case they are already recording motion before someone presses the doorbell, so you're not going to get an image then anyway.

Even if you configure the camera not to record on motion, my experience with images on ding is that they are not ideal.  You usually get an arm blocking most of the view, or just a torso and chin because they are too close to the camera.  Not that useful.  Anyone can just cover up the camera while they ring the doorbell if they don't want you to view, but this isn't true of motion.  I think, without the features of recording on motion, 95% of Ring customers wouldn't purchase the device to begin with.  I mean, Ring even markets the pre-roll feature heavily so you can see the time before motion was triggered.

**Reply by Pabsilon** [reply]
Thanks for your reply!
There's not any cut features on the ring app, it's just that if I were to activate those features I could be sued by every person recorded by my camera who doesn't ring. 
In general, you can't point a camera to the street outside of your house. A motion sensor on the other hand, would be fine, since you can't individually identify who triggered it. 
To my understanding, this is general EU law under the GDPR, but as I said, there are some exceptions, like someone ringing your door.

About the snapshot, I don't know how the ring app does it, but there's clearly a small delay, since every time someone has ringed my door their hand wasn't on the camera. 

If the stream is activated by the ring camera, could I watch it through rtsp and then take a snapshot of such stream, let's say, a second after it started? I haven't really been able to configure it..
